testsuite/tests/libratest/smoke_test.rs

```diff
@@ -336,3 +340,62 @@ fn test_full_node() {
     let response = client_proxy.mint_coins(&["mint", "0", "1"], false);
     assert!(response.is_err());
 }
+
+#[test]
+fn test_external_transaction_signer() {
+    let (_swarm, mut client_proxy) = setup_swarm_and_client_proxy(1, 0);
+
+    // generate key pair
+    let mut seed: [u8; 32] = [0u8; 32];
+    seed[..4].copy_from_slice(&[1, 2, 3, 4]);
+    let mut rng: StdRng = SeedableRng::from_seed(seed);
+    let key_pair = KeyPair::<Ed25519PrivateKey, Ed25519PublicKey>::generate_for_testing(&mut rng);
```

Comment:
I would recommend `let key_pair = compat::generate_keypair(None);` as in the background it uses a TEST seed if no rng is provided.

Reply:
Done.